packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -868,6 +868,12 @@ abstract class SliverMultiBoxAdaptorWidget extends SliverWithKeepAliveWidget {
 /// means [SliverList] cannot learn their main axis extent. Instead, newly
 /// materialized children are placed adjacent to existing children.
 ///
+/// If there exists some specific rules that the main axis extent of children
+/// can be calculated or reckoned out ahead of time, one can offer a
+/// [SliverCustomExtentListAssistant] to help [SliverList] to perform layout on
+/// its children. This gifts developer the abilit and responsibility to engage
+/// in promoting the efficiency of [SliverList].
```

Comment:
This still seems like a very special case that not many developers will experience, and as this states it puts a lot of the responsibility on the developer.
Often developers do not know the size of the items in a list, especially since there are a lot of varying screen sizes across platforms that their apps can be run on. Particularly on desktop for example, window size can be changed on the fly. Or in another case, I have often seen animation driven transformations as scrolling occurs.

Another consideration is that this would only be for those working with slivers. I agree with your statement above that this would be an advanced implementation option, but as such it would not actually solve the issues developers using ListView and higher level scrolling classes are experiencing. The performance issue is completely valid, but I don't know how many folks will find it resolved with this change.

I think this may be best served as a package on pub.dev as @goderbauer suggested above.